shopwild-backend/controllers/cart.js

```diff
@@ -0,0 +1,86 @@
+const Cart = require("../models/cart");
+
+exports.getAllCarts = async (req, res) => {
+  try {
+    const allCarts = await Cart.find()
+      .populate("user")
+      .populate("products.product");
+    res.json({ allCarts });
+  } catch (error) {
+    res.status(500).json({
+      message: "Unable to get carts",
+      errorMessage: error.message,
+    });
+  }
+};
+
+exports.getCart = async (req, res) => {
+  try {
+    const cart = await Cart.findOne({ user: req.user._id })
+      .populate("user")
+      .populate("products.product");
+    res.json({ cart });
+  } catch (error) {
+    res.status(404).json({
+      message: "Unable to get the cart for the user",
+      errorMessage: error.message,
+    });
+  }
+};
+
+exports.createUpdateCart = async (req, res) => {
+  try {
+    const cartUpdates = req.body;
+    let cart = await Cart.findOne({ user: req.user._id });
+
+    cartUpdates.user = req.user;
+
+    if (cart === null) {
```

Comment:
No need to compare with null,
```suggestion
    if (cart) {
```